functional-test/src/test/groovy/org/rundeck/tests/functional/api/jobs/JobScheduledSpec.groovy

```diff
@@ -0,0 +1,190 @@
+package org.rundeck.tests.functional.api.jobs
+
+import org.rundeck.util.annotations.APITest
+import org.rundeck.util.api.WaitingTime
+import org.rundeck.util.container.BaseContainer
+
+import java.time.LocalDateTime
+import java.time.ZoneId
+
+@APITest
+class JobScheduledSpec extends BaseContainer {
+
+    def setupSpec() {
+        startEnvironment()
+        setupProject()
+    }
+
+    def "scheduled job run should succeed (sleep 20 sec)"(){
+        setup:
+            def path = generatePath()
+            def jobId = jobImportFile(path).succeeded[0].id
+        when:
+            def response = doGet("/job/${jobId}/executions?status=succeeded")
+            def count = jsonValue(response.body()).executions.size()
+        then:
+            verifyAll {
+                response.successful
+                response.code() == 200
+            }
+        when:
+            sleep 20000
+            def response1 = doGet("/job/${jobId}/executions?status=succeeded")
+        then:
+            verifyAll {
+                response1.successful
+                response1.code() == 200
+                def json = jsonValue(response1.body())
+                def count2 = json.executions.size()
+                def testVal = count + 1
+                testVal == count2 && count != null && count2 != null
+            }
+    }
+
+    def "job/id/run should succeed"() {
+        setup:
+            def path = generatePath1()
+            def jobId = jobImportFile(path).succeeded[0].id
+        when:
+            def execId = runJob(jobId, ["options":["opt2": "a"]])
+            sleep WaitingTime.MODERATE.milliSeconds
+            def responseExec = get("/execution/${execId}", Map)
+        then:
+            verifyAll {
+                responseExec.status == 'timedout'
+            }
+    }
+
+    def "job/id/run should succeed with retry"() {
```

Comment:
This test fails locally, execution state stills running

Reply:
why not to use "JobUtils.waitForExecutionToBe"?

Reply:
Change is made to use that method.